R/query-string.R

```diff
@@ -11,116 +11,131 @@ queryStringFilter <- function(req){
 
 #' @noRd
 parseQS <- function(qs){
-  if (is.null(qs) || length(qs) == 0 || qs == "") {
+
+  if (is.null(qs) || length(qs) == 0L || qs == "") {
     return(list())
   }
-  if (stri_startswith_fixed(qs, "?")) {
-    qs <- substr(qs, 2, nchar(qs))
-  }
 
-  parts <- strsplit(qs, "&", fixed = TRUE)[[1]]
-  kv <- strsplit(parts, "=", fixed = TRUE)
-  kv <- kv[vapply(kv, length, numeric(1)) == 2] # Ignore incompletes
+  qs <- stri_replace_first_regex(qs, "^[?]", "")
+  #qs <- stri_replace_all_fixed(qs, "+", " ")
+
+  args <- stri_split_fixed(qs, "&", omit_empty = TRUE)[[1L]]
+  kv <- lapply(args, function(x) {
+    # returns utf8 strings
+    httpuv::decodeURIComponent(stri_split_fixed(x, "=", omit_empty = TRUE)[[1]])
+  })
+  kv <- kv[lengths(kv) == 2] # Ignore incompletes
 
   if (length(kv) == 0) {
     # return a blank list of args if there is nothing to parse
     return(list())
   }
 
-  keys <- httpuv::decodeURIComponent(vapply(kv, "[[", character(1), 1)) # returns utf8 strings
-  if (any(Encoding(keys) != "unknown")) {
+  k <- vapply(kv, "[", character(1), 1)
+  kenc <- unique(Encoding(k))
```

Comment:
How come we are no longer decoding the URI?

Reply:
It is done on line 25 above,